tools/contrib/subset_data.py

```diff
@@ -0,0 +1,1096 @@
+                    required=False,
+                    type = plon_type,
+                    default= 287.8 )
+        pt_parser.add_argument('--site',
+                    help='Site name or tag. [default: %(default)s]', 
+                    action="store",
+                    dest="site_name",
+                    required = False,
+                    type = str, 
+                    default = '')
+        pt_parser.add_argument('--create_domain',
+                    help='Flag for creating CLM domain file at single point. [default: %(default)s]', 
+                    action="store_true",
+                    dest="create_domain",
+                    required = False,
+                    default = False)
+        pt_parser.add_argument('--create_surface',
+                    help='Flag for creating surface data file at single point. [default: %(default)s]', 
+                    action="store",
+                    dest="create_surfdata",
+                    required = False,
+                    type = bool, 
+                    default = True)
+        pt_parser.add_argument('--create_landuse',
+                    help='Flag for creating landuse data file at single point. [default: %(default)s]', 
+                    action="store_true",
+                    dest="create_landuse",
+                    required = False,
+                    default = False)
+        pt_parser.add_argument('--create_datm',
+                    help='Flag for creating DATM forcing data at single point. [default: %(default)s]', 
+                    action="store_true",
+                    dest="create_datm",
+                    required = False,
+                    default = False)
+        pt_parser.add_argument('--datm_syr',
+                    help='Start year for creating DATM forcing at single point. [default: %(default)s]', 
+                    action="store",
+                    dest="datm_syr",
+                    required = False,
+                    type = int, 
+                    default = 1901)
+        pt_parser.add_argument('--datm_eyr',
+                    help='End year for creating DATM forcing at single point. [default: %(default)s]', 
+                    action="store",
+                    dest="datm_eyr",
+                    required = False,
+                    type = int,
+                    default = 2014)
+        pt_parser.add_argument('--crop', 
+                    help='Create datasets using the extensive list of prognostic crop types. [default: %(default)s]', 
+                    action="store_true", 
+                    dest="crop_flag", 
+                    default=False)
+        pt_parser.add_argument('--dompft', 
+                    help='Dominant PFT type . [default: %(default)s] ', 
+                    action="store", 
+                    dest="dom_pft",
+                    type =int,
+                    default=7)
+        pt_parser.add_argument('--no-unisnow', 
```

Comment:
@wwieder Thanks so much for this comment. 
I essentially changed this at the last minute to fix an issue with python boolean parsing. 
Basically, in python, if the argument type is bool the parsing does not work out-of-box as expected, meaning it considers both False and True as True.

I just wrote this simple code to demonstrate it:

```
import argparse
parser = argparse.ArgumentParser(description="Parse bool")
parser.add_argument("--foo", action="store", type =bool, dest = "foo")
args = parser.parse_args()
print (args.foo)
```

Here if we run, the output is true non-the less.
```
>> python test.py --foo false
True
```

And even if you additionally specify choices=[False,True], you end up with both "False" and "True" considered True.

On another level, even something like this will ouput True in python:

```
>>> bool("False")
True
```
-----------------------
In python, to resolve this issue, they usually use `--feature` option to turn on a feature and `--no-feature` to turn off a feature. 
Here for the options that that were on (True) by default, I defined --no*** to turn that option off.
For example, here `uni_snow` is by default `True`; therefore, I defined a command-line option to turn it to `False` if the user need it as `False`:


https://github.com/ESCOMP/CTSM/blob/b06af57e1ed44773a297fe73a45bb84afc270665/tools/contrib/subset_data.py#L181-L185

Reply:
That being said, it seems like in Python 3.9 (late 2020) and above they have introduced this capability to pass boolean directly as command-line argument, using `BooleanOptionalAction` in `argparse`. 

But in Cheyenne we have python/3.7.9 as the latest version of python available...

Reply:
OK now I think I understand.  Let's make the help statement here more informative so that if a user defines --no-unisnow they will create a non-uniform snowpack (but the default (True) is for a uniform snowpack).

Reply:
To address this comment, we decided to:
- [ ] Add some clarification for the `--no-feature` flags for example adding "Flag for turning off the feature option." 
I added these clarifications in commit b854ad6.